### PR TITLE
Fix rsa sign verify

### PR DESCRIPTION
**Pull request by bandi13**
This is a proposed fix to ZD#18157.

### CODE REVIEW

**Review by kareem-wolfssl — DISMISSED** [review]

File: src/wp_rsa_sig.c [hunk 1]
```diff
@@ -414,15 +414,19 @@ static int wp_rsa_signverify_init(wp_RsaSigCtx* ctx, wp_Rsa* rsa,
 {
     int ok = 1;
 
-    if (ctx->rsa != rsa) {
+    if ((ctx == NULL) || (rsa == NULL)) {
+        ok = 0;
+    }
+    if ((ok) && (ctx->rsa != rsa)) {
```
Comment (SparkiDev) [inline src/wp_rsa_sig.c:420]: No need for brackets around ok.

File: src/wp_rsa_sig.c [hunk 2]
```diff
@@ -414,15 +414,19 @@ static int wp_rsa_signverify_init(wp_RsaSigCtx* ctx, wp_Rsa* rsa,
...
         wp_rsa_free(ctx->rsa);
         ctx->rsa = NULL;
         if (!wp_rsa_up_ref(rsa)) {
             ok = 0;
+        } else {
```
Comment (SparkiDev) [inline src/wp_rsa_sig.c:425]: Formatting:
```
}
else {
```

File: src/wp_rsa_sig.c [hunk 3]
```diff
@@ -432,14 +436,14 @@ static int wp_rsa_signverify_init(wp_RsaSigCtx* ctx, wp_Rsa* rsa,
     if (ok) {
         ctx->saltLen = RSA_PSS_SALTLEN_AUTO;
 
-        if (wp_rsa_get_type(rsa) == RSA_FLAG_TYPE_RSA) {
+        if (wp_rsa_get_type(ctx->rsa) == RSA_FLAG_TYPE_RSA) {
```
Comment (SparkiDev) [inline src/wp_rsa_sig.c:439]: No need for this. ctx->rsa == rsa at this point and on.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/wp_rsa_sig.c (hunk 3 above)
Comment (bandi13) [reply src/wp_rsa_sig.c:439]: As I mentioned in the commit message, this is just a defensive coding. Currently `ctx->rsa == rsa`, but there may be a future change above that inadvertently violates this. Additionally, this doesn't use too much more code space.

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**